From add1e4f2f4ea6f8551ceac7ebe5602c37b30d594 Mon Sep 17 00:00:00 2001
From: Wendy Liang <wendy.liang@xilinx.com>
Date: Thu, 16 Jan 2020 13:18:42 -0800
Subject: [PATCH 0854/1852] remoteproc: sysfs: allow kick to pass small data

commit 9f861656e53e4b59040bb05cc14b4b7bd4a0ed2b from
https://github.com/Xilinx/linux-xlnx.git

Allow kick to pass small data. The size of the data to pass by sysfs
kick depends on the kick rproc operation implementation. For now, the
size is sizeof(int).

Signed-off-by: Wendy Liang <wendy.liang@xilinx.com>
Reviewed-by: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/remoteproc/remoteproc_sysfs.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/remoteproc/remoteproc_sysfs.c b/drivers/remoteproc/remoteproc_sysfs.c
index ed89bdac1da5..e680d32ba89f 100644
--- a/drivers/remoteproc/remoteproc_sysfs.c
+++ b/drivers/remoteproc/remoteproc_sysfs.c
@@ -130,12 +130,15 @@ static ssize_t kick_store(struct device *dev,
 			  const char *buf, size_t count)
 {
 	struct rproc *rproc = to_rproc(dev);
+	int id;
+	size_t cpy_len;
 
 	(void)attr;
-	(void)buf;
+	cpy_len = count <= sizeof(id) ? count : sizeof(id);
+	memcpy((char *)(&id), buf, cpy_len);
 
 	if (rproc->ops->kick)
-		rproc->ops->kick(rproc, 0);
+		rproc->ops->kick(rproc, id);
 	else
 		count = -EINVAL;
 	return count;
-- 
2.31.1

